lib/active_merchant/billing/gateway.rb

```diff
@@ -57,7 +57,7 @@ class Gateway
       include CreditCardFormatting
 
       DEBIT_CARDS = [ :switch, :solo ]
-      CURRENCIES_WITHOUT_FRACTIONS = %w(BIF BYR CLP CVE DJF GNF HUF ISK JPY KMF KRW PYG RWF UGX VND VUV XAF XOF XPF)
+      CURRENCIES_WITHOUT_FRACTIONS = %w(BIF BYR CLP CVE DJF GNF HUF IDR ISK JPY KMF KRW PYG RWF UGX VND VUV XAF XOF XPF)
```

Comment:
I'm not sure `IDR` should apply to all gateways.  I wonder if WorldpayGateway should have its own list of `CURRENCIES_WITHOUT_FRACTIONS` like Stripe does.